core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/Collector.scala

```diff
@@ -155,7 +159,11 @@ trait Collector {
         Http()
           .newServerAt(collectorConf.interface, collectorConf.ssl.port)
           .enableHttps(ConnectionContext.httpsServer(SSLContext.getDefault)),
-        routes
+        optionalHeaderValueByName("X-Forwarded-Proto") {
+          case Some(clientProtocol) if clientProtocol == "http" =>
+            redirectTo(Scheme.Http, collectorConf.port, StatusCodes.MovedPermanently)
+          case _ => routes
+        }
```

Comment:
This `redirectTo` should only get called if the user enabled `collector.ssl.redirect` in the hocon.  Please can you check whether the config parameter is obeyed? I cannot see where the check is done.

Reply:
why do we need `collector.ssl.redirect` check for this? if `X-Forwarded-Proto: http` header exists and LB sent it to https port, I'd expect collector to respect the header and route the request to http port, regardless of `collector.ssl.redirect` setting as it is about redirecting http to https

Reply:
If `X-Forwarded-Proto: http` header exists, it means the tracker sent the request to the load balancer's http port.

If hocon has `collector.ssl.redirect = false` it means the pipeline operator is happy to accept tracker requests sent to the load balancer's http port.  So we should sink the request immediately and not issue a redirect.

If hocon has `collector.ssl.redirect = true` it means the pipeline operator does not want trackers to send requests to the load balancer's http port, for security reasons.  So the collector should return a `MovedPermanently` to make the tracker send the request to the load balancer's port.

To be clear.... the status code we return here is an instruction to the tracker on what to do next, it is not an instruction to the load balancer.